### PR TITLE
Done changes as per review comments from Asir

### PR DESCRIPTION
**Pull request by anudeepsharma**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anudeepsharma__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ans). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: azure-mgmt-sql/src/main/java/com/microsoft/azure/management/sql/SqlDatabase.java [hunk 1]
```diff
@@ -122,6 +123,7 @@
     /**
      * @return SqlWarehouse instance for more operations
      */
+    @Method
     SqlWarehouse castToWarehouse();
```
Comment (martinsawicki) [inline azure-mgmt-sql/src/main/java/com/microsoft/azure/management/sql/SqlDatabase.java]: the usual naming convention for this sort of thing would be `asWarehouse()`

**Comment by martinsawicki** [comment]
adding a compat-breaks tag - presumably the enhanced API design is not compatible with beta4 right?
If so, pls elaborate what changed exactly in the PR comment.

File: azure-mgmt-sql/src/main/java/com/microsoft/azure/management/sql/SqlDatabase.java [hunk 2]
```diff
@@ -314,6 +330,22 @@
...
+            /**
+             * Sets the max size in bytes for SQL Database.
+             *
+             * @param maxSizeBytes max size of the Azure SQL Database expressed in bytes. Note: Only
+             * the following sizes are supported (in addition to limitations being
+             * placed on each edition): { 100 MB | 500 MB |1 GB | 5 GB | 10 GB | 20
+             * GB | 30 GB … 150 GB | 200 GB … 500 GB }
+             * @return The next stage of the definition.
+             */
+            WithCreateWithElasticPoolOptions withMaxSizeBytes(long maxSizeBytes);
```
Comment (anuchandy) [inline azure-mgmt-sql/src/main/java/com/microsoft/azure/management/sql/SqlDatabase.java]: May be in future, in these type of scenarios, where user needs to do some math for conversion (here nGB to equivalent  bytes, mMB to equivalent bytes etc), we can think of having an overload like:

```java
withMaxSizeBytes(long size, SizeUnit unit)

where SizeUnit is an enum { GB, MB, KB, Byte }.
```

See similar example here    http://reactivex.io/RxJava/javadoc/rx/Observable.html#delay(long,%20java.util.concurrent.TimeUnit)

```java
Observable<T> delay(long delay, java.util.concurrent.TimeUnit unit)
```

**Comment by anudeepsharma** [comment]
Made changes to remove withoutElasticPool and withoutSourceDatabase from the Database creation.

**Pull request merged**